### ISSUE TITLE
Should be possible to customize model transformation (Transformation is not @ChainOverridable)

### ISSUE THREAD

**Issue by kamaladafrica** [sculptor/sculptor#64]
In the version before, this was done through AOP advices in the file src/main/resources/extensions/SpecialCases.ext, like in the following snippet:
```
import sculptormetamodel;

around transformation::Transformation::modifyUuid(DomainObject domainObject) :
    null;
```

**Comment by tjuerge** [comment]
Using `@ChainOverridable` on Xtend classes which are maintaining state (explicitly via member variables or implicitly via the create methods cache implementation) leads to strange error like this. This is due to the inheritance hierarchy created by Sculptors active annotation processor and maintaining separate instances: The instance of the overriding class has it's state (via extending the class annotated with `@ChainOverridable`) and the instance of the overridable class has it's state. Calling methods on the overriding class or the overridable class manipulates the state within separate instances of the annotated class.

**Comment by tavoda** [comment]
After all this troubles it is maybe time to change implementation. Maybe get rid of annotation processor (or at least extensive manipulation we use) and use some kind of proxy and IoC manipulations. What do you think?

**Comment by tavoda** [comment]
Somebody took over our bug in Xtend:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=446364
https://github.com/eclipse/xtext/pull/43

Maybe we will be able to override also 'create' methods finally.

**Comment by tjuerge** [comment]
But this doesn't solve our issues with the cache implementation generated for Xtends create methods (as mentioned by Jan Koehnlein in [his comment](https://bugs.eclipse.org/bugs/show_bug.cgi?id=446364#c1)):
Every extension class gets its own instances of these caches. There's no global cache maintained across the chained extensions and the corresponding overriden template.

So it seems that we have to come up with our own cache implementation for create methods, eg. within the chain link code.

**Comment by tavoda** [comment]
Yes I was playing with this, however I always found obstacle to finish it. That's why I filled this ticket on Xtend, 3 different errors always fired. Now 'maybe' we can find way to override create methods. It was impossible without this fix.
